src/stores/RouterStore.ts

```diff
@@ -90,11 +93,18 @@ export class RouterStore {
      *     Options for stringifying query params. These are passed directly to
      *     the [query-string](https://github.com/sindresorhus/query-string)
      *     library that is used internally.
+     *
+     *   globalBeforeEnter: function
+     *     A global routing guard applied to each page transition
+     *     //!@# TBD
      */
     constructor(
         routes: Route[],
         notFoundState: RouterState,
-        options: { [key: string]: any } = {}
+        options: {
+            [key: string]: any,
+            globalBeforeEnter?: TransitionHook,
```

Comment:
I like this idea, but I don't know how to make it work in general. I was thinking of creating a separate interface called RouterOptions which defines all known options and a place to add additional ones. For example this, but was not able to make it work:

export interface RouterOptions {
    initialState?: RouterState;
    queryParseOptions?: any;
    queryStringifyOptions?: any;
    globalBeforeEnter?: TransitionHook,
    [key: string]: any,
}

See if you can make this work, otherwise take this line out and assume that globalBeforeEnter is an optional key that caller can send in. This is not going to be tightly typed, but at least it is consistent with what exists currently.